src/gc/unix/cgroup.cpp

```diff
@@ -289,28 +427,43 @@ class CGroup
                     goto done;
                 maxLineLen = lineLen;
             }
-                   
-            // See man page of proc to get format for /proc/self/cgroup file
-            int sscanfRet = sscanf(line, 
-                                   "%*[^:]:%[^:]:%s",
-                                   subsystem_list,
-                                   cgroup_path);
-            if (sscanfRet != 2)
+
+            if (is_cgroupv2)
             {
-                assert(!"Failed to parse cgroup info file contents with sscanf.");
-                goto done;
+                // See https://www.kernel.org/doc/Documentation/cgroup-v2.txt
+                // Look for "0::/some/path"
+                int sscanfRet = sscanf(line,
+                                       "0::%s",
+                                       cgroup_path);
+                if (sscanfRet == 1)
+                {
+                    result = true;
+                }
             }
-    
-            char* context = nullptr;
-            char* strTok = strtok_r(subsystem_list, ",", &context); 
-            while (strTok != nullptr)
+            else
             {
-                if (is_subsystem(strTok))
+                // See man page of proc to get format for /proc/self/cgroup file
+                int sscanfRet = sscanf(line,
+                                       "%*[^:]:%[^:]:%s",
+                                       subsystem_list,
+                                       cgroup_path);
```

Comment:
Ditto.

Reply:
Good catch. But I only moved this code around. Is it better to fix it up now or do that cleanup in a separate change?

Reply:
It's usually a good idea to leave the code better than you found it.